src/tiled/editableobject.cpp

```diff
@@ -71,6 +73,63 @@ void EditableObject::removeProperty(const QString &name)
         mObject->removeProperty(name);
 }
 
+QVariantMap EditableObject::propertiesIncludingInherited() const
+{
+    auto mergeVariantMaps = [](QVariantMap& dstMap, const QVariantMap& srcMap) {
+        for (auto iter = srcMap.constBegin(); iter != srcMap.constEnd(); ++iter)
+        {
+            dstMap.insert(iter.key(), iter.value());
+        }
+    };
+
+    QVariantMap allProperties;
+    // Insert properties into allProperties in the reverse order that
+    // Object::inheritedProperty searches them, to make sure that the
+    // same precedence is maintained.
+
+    QString objectType;
+    switch (mObject->typeId()) {
+    case Object::MapObjectType:
+    {
```

Comment:
Coding style: please put the `{` on the same line.

Reply:
Old habits die hard.